cmd/infra/powervs/destroy.go

```diff
@@ -371,8 +376,64 @@ func deleteSecrets(name, namespace, cloudInstanceID string, accountID string, re
 	return nil
 }
 
+// destroyPowerVsDhcpServer destroying powervs dhcp server
+func destroyPowerVsDhcpServer(ctx context.Context, logger logr.Logger, cloudInstanceID string, session *ibmpisession.IBMPISession) error {
+	client := instance.NewIBMPIDhcpClient(ctx, session, cloudInstanceID)
+
+	dhcpServers, err := client.GetAll()
+	if err != nil {
+		return err
+	}
+
+	if dhcpServers == nil || len(dhcpServers) < 1 {
+		logger.Info("No DHCP servers available to delete in PowerVS")
+		return nil
+	}
+
+	dhcpID := *dhcpServers[0].ID
```

Comment:
Is there a possibility that there are multiple dhcp servers in a workspace ? We may have to iterate through all of it and delete.

Reply:
No, there will be only one dhcp server.
https://github.com/openshift/hypershift/blob/main/cmd/infra/powervs/create.go#L130